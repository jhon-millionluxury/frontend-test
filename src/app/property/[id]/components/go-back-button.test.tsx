/* eslint-disable @typescript-eslint/no-explicit-any */
import { render, screen } from "@testing-library/react";
import PropertyDetailPageGoBackButton from "@/app/property/[id]/components/go-back-button";

// Mock de next/link
jest.mock("next/link", () => {
  const Link = ({ children, href }: any) => (
    <a href={href} data-testid="link">
      {children}
    </a>
  );
  Link.displayName = "MockNextLink";
  return Link;
});

// Mock de Button
jest.mock("../../../../components/ui/button", () => ({
  Button: ({ children, ...props }: any) => (
    <button data-testid="button" {...props}>
      {children}
    </button>
  ),
}));

// Mock de ArrowLeft (Ã­cono)
jest.mock("lucide-react", () => ({
  ArrowLeft: (props: any) => <svg data-testid="icon" {...props} />,
}));

describe("PropertyDetailPageGoBackButton", () => {
  it("renders the button text correctly", () => {
    render(<PropertyDetailPageGoBackButton />);
    expect(screen.getByText("Back to Properties")).toBeInTheDocument();
  });

  it("wraps the button inside a Link to home", () => {
    render(<PropertyDetailPageGoBackButton />);
    const link = screen.getByTestId("link");
    expect(link).toHaveAttribute("href", "/");
    expect(screen.getByTestId("button")).toBeInTheDocument();
  });

  it("renders the ArrowLeft icon", () => {
    render(<PropertyDetailPageGoBackButton />);
    expect(screen.getByTestId("icon")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<PropertyDetailPageGoBackButton />);
    expect(asFragment()).toMatchSnapshot();
  });
});
