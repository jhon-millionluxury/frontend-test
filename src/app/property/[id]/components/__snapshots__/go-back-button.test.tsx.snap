// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`PropertyDetailPageGoBackButton matches snapshot 1`] = `
<DocumentFragment>
  <a
    data-testid="link"
    href="/"
  >
    <button
      class="mb-6"
      data-testid="button"
      variant="ghost"
    >
      <svg
        class="mr-2 h-4 w-4"
        data-testid="icon"
      />
      Back to Properties
    </button>
  </a>
</DocumentFragment>
`;
